core/src/main/scala/com/yahoo/maha/core/dimension/Dimension.scala

```diff
@@ -159,6 +165,37 @@ object HiveDerDimCol {
   }
 }
 
+case class HiveDerDimAggregateCol(name: String,
+                         dataType: DataType,
+                         columnContext: ColumnContext,
+                         derivedExpression: HiveDerivedExpression,
+                         alias: Option[String],
+                         annotations: Set[ColumnAnnotation],
+                         filterOperationOverrides: Set[FilterOperation]) extends BaseDerivedAggregateDimCol with WithHiveEngine {
+
+  require(derivedExpression != null,
+    s"Derived expression should be defined for a derived column $name")
+  require(derivedExpression.expression.hasRollupExpression, s"HiveDerDimAggregateCol should have rollup expression  $name - $derivedExpression")
```

Comment:
You need to ensure the expression only references other dim cols, not fact cols.

Reply:
yup , very good point. I will add validation.

Reply:
added validation